rust_book_ex_8_3/src/string_ex.rs

```diff
@@ -0,0 +1,21 @@
+fn starts_with_vowel(s: &String) -> bool {
+    let vowels = vec!['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
+
+    if let Some(first_char) = s.chars().next() {
```

Comment:
I think the logic is correct, but it's a bit verbose :) You could do something like

```rust
const VOWELS: &'static str = "aeiouAEIOU";
s.chars().next().map(|c| VOWELS.contains(c)).unwrap_or(false)
```

Reply:
Alternatively, you could make use of `.to_ascii_lowercase()`:

```rust
    const VOWELS: &'static str = "aeiou";
    s.chars().next()
        .map(|c| VOWELS.contains(c.to_ascii_lowercase())).unwrap_or(false)
```